### PR TITLE
Deprecate toString() in favor of __toString()

### PR DESCRIPTION
**Pull request by deguif**

### CODE REVIEW

File: tests/BulkTest.php [hunk 1]
```diff
@@ -374,19 +374,22 @@ public function testRawDocumentDataRequest(): void
...
-        $bulk = new Bulk($client);
-        $bulk->addDocuments($documents);
-        $bulk->setIndex($index);
+        $bulk = (new Bulk($client))
+            ->setIndex($index)
+            ->addDocuments($documents)
+        ;
 
-        $expectedJson = '{"index":{}}
+        $expectedJson = <<<'JSON'
```
Comment (ruflin) [inline tests/BulkTest.php:382]: What is the `JSON` used here for? Haven't seen this before.

**Review by ruflin — APPROVED** [review]
++ on getting rid of the toString() methods and instead use the magic methods.

File: tests/BulkTest.php (hunk 1 above)
Comment (deguif) [reply tests/BulkTest.php:382]: That's just HEREDOC syntax with `JSON` identifier ;)

Comment (ruflin) [reply tests/BulkTest.php:382]: Ah, got it 👍

**Pull request merged**